Домашнее задание №8

MAMKUH6OPIIIUK opened the pull request:
Выполнил ДЗ №8.
Относительно предыдущего ДЗ немного изменена модель - жанры вкладываются в документы книг и не имеют id-шников (ну или имя жанра = естественный ключ). Соответственно, от репозитория жанров остался только кастомный репозиторий с одним методом findAll, получающим все уникальные жанры книг при помощи агрегации. 
Остальное всё реализовано при помощи отдельных коллекций со связями между ними. 
Для тестов используется flapdoodle, для "прода" - настоящая MongoDB. БД постарался назвать так, чтобы ни с кем точно не пересекалась 😄

MAMKUH6OPIIIUK committed:
ДЗ №8. Initial commit

MAMKUH6OPIIIUK committed:
Выполнил домашнюю работу №8

vlad-iv commented on spring-hw-08/pom.xml:
```diff
@@ -0,0 +1,135 @@
+        <dependency>
+            <groupId>org.projectlombok</groupId>
+            <artifactId>lombok</artifactId>
+            <version>${lombok.version}</version>
```
Можно не указывать

MAMKUH6OPIIIUK replied:
Спасибо! Проморгал :-(

vlad-iv commented on spring-hw-08/pom.xml:
```diff
@@ -0,0 +1,135 @@
+        <artifactId>spring-boot-starter-parent</artifactId>
+        <version>3.0.5</version>
+        <relativePath/> <!-- lookup parent from repository -->
+    </parent>
```
Spring boot можно перенести в корневой (что бы не указывать зависимости в каждом из проектов)

MAMKUH6OPIIIUK replied:
Перенес spring-boot-parent в корневой pom, заодно добавил управление зависимостями, указание некоторых общих зависимостей и управление сборкой туда же.
Рефакторнул все помники с 3 по 8 ДЗ - теперь у всех их родителем является корневой pom, и убрано всё лишнее.

MAMKUH6OPIIIUK committed:
Рефакторинг управления зависимостями и сборкой ДЗ №3-8:

1. Все версии зависимостей, которыми не управляет spring-boot-parent, теперь настраиваются в корневом pom
2. В корневом pom теперь сразу указаны общие зависимости, чтобы их не нужно было указывать в дочерних модулях
3. В корневом pom теперь полностью настроена общая сборка дочерних модулей, чтобы одинаковую секцию build не нужно было указывать в дочерних модулях.
